cirq/ops/common_gates.py

```diff
@@ -438,13 +438,11 @@ class MeasurementGate(raw_types.Gate):
     of measurements.
     """
 
-    def num_qubits(self) -> int:
-        return self._num_qubits
-
     def __init__(self,
                  num_qubits: int,
```

Comment:
You can make this argument default to `None`, so that it's not necessayr to specify it if qid shape is specified.

Reply:
Added.  It makes the init code a bit more complicated.